File: solutions/python/cricinfo/scorecard.py
class Scorecard:
    def __init__(self, id, match):
        self.id = id
        self.match = match
        self.team_scores = {}
        self.innings = []

    def update_score(self, team_id, score):
        self.team_scores[team_id] = score

    def add_innings(self, innings):
        self.innings.append(innings)

    def get_id(self):
        return self.id

    def get_match(self):
        return self.match

    def get_team_scores(self):
        return self.team_scores

    def get_innings(self):
        return self.innings

File: solutions/java/src/cricinfo/Scorecard.java
package cricinfo;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Scorecard {
    private final String id;
    private final Match match;
    private final Map<String, Integer> teamScores;
    private final List<Innings> innings;

    public Scorecard(String id, Match match) {
        this.id = id;
        this.match = match;
        this.teamScores = new HashMap<>();
        this.innings = new ArrayList<>();
    }

    public void updateScore(String teamId, int score) {
        teamScores.put(teamId, score);
    }

    public void addInnings(Innings innings) {
        this.innings.add(innings);
    }

    public String getId() {
        return id;
    }

    public Match getMatch() {
        return match;
    }

    public Map<String, Integer> getTeamScores() {
        return teamScores;
    }

    public List<Innings> getInnings() {
        return innings;
    }
}
